File: dp/53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        
        ret = -sys.maxsize
        cur = 0 
        for i in nums:
            cur = cur + i if cur > 0 else i
            ret = ret if ret > cur else cur
        return ret

File: dp/53.java
class Solution {
    public int maxSubArray(int[] nums) {
        int ret = Integer.MIN_VALUE;
        int cur = 0;
        for (int i : nums) {
            if (cur < 0) {
                cur = i;
            } else {
                cur += i;
            }
            if (ret < cur) {
                ret = cur;
            }
        }
        return ret;
    }
}
